### PR TITLE
Removed AppError from searchlayer/user_layer.go

### PR DESCRIPTION
**Pull request by SimonSimonB**
#### Summary
Removes `model.AppError` from searchlayer/user_layer.go.

#### Ticket Link
Fixes [#16418](https://github.com/mattermost/mattermost-server/issues/16418)

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by agnivade** [comment]
@hanzei - this is still in draft. I am in touch with @SimonSimonB to fix some issues. I will do the review once those are fixed.

File: store/searchlayer/user_layer.go [hunk 1]
```diff
@@ -168,13 +167,7 @@ func (s *SearchUserStore) getListOfAllowedChannelsForTeam(teamId string, viewRes
...
 		channels, err := s.rootStore.Channel().GetTeamChannels(teamId)
 		if err != nil {
-			var nfErr *store.ErrNotFound
-			switch {
-			case errors.As(err, &nfErr):
-				return nil, model.NewAppError("getListOfAllowedChannelsForTeam", "app.channel.get_channels.not_found.app_error", nil, nfErr.Error(), http.StatusNotFound)
-			default:
-				return nil, model.NewAppError("getListOfAllowedChannelsForTeam", "app.channel.get_channels.get.app_error", nil, err.Error(), http.StatusInternalServerError)
-			}
+			return nil, err
```
Comment (rvillablanca) [inline store/searchlayer/user_layer.go:170]: Can you add more context here using `errors.Wrap` in the same way as line 181?

File: store/searchlayer/user_layer.go [hunk 2]
```diff
@@ -185,7 +178,7 @@ func (s *SearchUserStore) getListOfAllowedChannelsForTeam(teamId string, viewRes
 	if len(viewRestrictions.Channels) > 0 {
 		channels, err := s.rootStore.Channel().GetChannelsByIds(viewRestrictions.Channels, false)
 		if err != nil {
-			return nil, model.NewAppError("getListOfAllowedChannelsForTeam", "app.channel.get_channels_by_ids.app_error", nil, err.Error(), http.StatusInternalServerError)
+			return nil, errors.Wrapf(err, "Failed to get channels by ids")
```
Comment (rvillablanca) [inline store/searchlayer/user_layer.go:181]: It is not required to use `errors.Wrapf` as you are not passing additional values, `errors.Wrap` is enough :)

Additionally, by convention go errors are in lowercase [(here)](https://github.com/golang/go/wiki/CodeReviewComments#error-strings). I know we have several places with capitalized errors but you can have your PR following best practices and conventions of go

**Review by rvillablanca — CHANGES_REQUESTED** [review]
I left a couple of minor suggestions but all is looking good. Thank you for your contribution @SimonSimonB.

File: store/searchlayer/user_layer.go (hunk 1 above)
Comment (SimonSimonB) [reply store/searchlayer/user_layer.go]: I changed this to `return nil, errors.Wrap(err, "failed to get team channels")` to provide context.

File: store/searchlayer/user_layer.go (hunk 2 above)
Comment (SimonSimonB) [reply store/searchlayer/user_layer.go]: I changed `errors.Wrapf(...)` to `errors.Wrap(...)` and changed the error messages to lowercase.

**Review by rvillablanca — APPROVED** [review]
Looks good. Thanks for your work

**Review by agnivade — APPROVED** [review]
Amazing work @SimonSimonB !

**Pull request merged**